crates/shrs_core/src/builtin/type.rs

```diff
@@ -0,0 +1,127 @@
+use super::BuiltinCmd;
+
+use anyhow::Error;
+use clap::Parser;
+
+use crate::{
+    prelude::CmdOutput,
+    shell::{Context, Runtime, Shell},
+};
+
+use std::fs::metadata;
+
+#[derive(Parser)]
+struct Cli {
+    #[arg(short = 'P')]
```

Comment:
long flags for each of these and also some docs would be nice (we don't need to follow basic type builtin exactly)